src/gui/widgets/SideBarWidget.cpp

```diff
@@ -22,23 +22,34 @@
  *
  */
 
+#include "SideBarWidget.h"
+
 #include <QApplication>
 #include <QFontMetrics>
 #include <QPainter>
 
-#include "SideBarWidget.h"
+#include "embed.h"
 
 
 SideBarWidget::SideBarWidget( const QString & _title, const QPixmap & _icon,
 							QWidget * _parent ) :
 	QWidget( _parent ),
 	m_title( _title ),
-	m_icon( _icon )
+	m_icon(_icon),
+	m_buttonSize(17, 17)
 {
 	m_contents = new QWidget( this );
 	m_layout = new QVBoxLayout( m_contents );
 	m_layout->setSpacing( 5 );
 	m_layout->setMargin( 0 );
+	m_closeBtn = new QPushButton(embed::getIconPixmap("close"), QString(), this);
+	m_closeBtn->resize(m_buttonSize);
+	m_closeBtn->setFocusPolicy(Qt::NoFocus);
+	m_closeBtn->setCursor(Qt::ArrowCursor);
+	m_closeBtn->setAttribute(Qt::WA_NoMousePropagation);
```

Comment:
Are there any reasons for setting `Qt::NoFocus` and `Qt::WA_NoMousePropagation`?

Reply:
It seems like `Qt::NoFocus` and `Qt::ArrowCursor` are set by default. I think the `SubWindow` needs `Qt::WA_NoMousePropagation` because it's undesirable to propagate mouse events of the button to the parent subwindow. It's fine to either keep or drop the line because mouse events on `SidebarWidget` are no-ops.

Reply:
I'll remove them for simplicity.